receiver/dotnetdiagnosticsreceiver/dotnet/parser.go

```diff
@@ -44,3 +46,71 @@ func (p *Parser) ParseIPC() error {
 func (p *Parser) ParseNettrace() error {
 	return parseNettrace(p.r)
 }
+
+// ParseAll parses all of the blocks until an error occurs or the context is
+// cancelled.
+func (p *Parser) ParseAll(ctx context.Context) error {
+	var err error
+	fms := fieldMetadataMap{}
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+			err = p.parseBlock(fms)
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
+
+// parseBlock parses a serialization type and block, bookended by a begin and
+// end object tag.
+func (p *Parser) parseBlock(fms fieldMetadataMap) error {
+	err := beginPrivateObject(p.r)
+	if err != nil {
+		return err
+	}
+
+	st, err := parseSerializationType(p.r)
+	if err != nil {
+		return err
+	}
+
+	p.logger.Info("parsing block", zap.String("serialization type", st.name))
```

Comment:
Does this need to be `Info`? Would it not flood the logs?

Reply:
It's max 1 log message per second, but the information is probably not terribly useful except for debugging. Changed to `Debug`.